### PR TITLE
Block Supports: Add link hover color to individual blocks

### PR DESCRIPTION
**Pull request by youknowriad**
This back ports the php changes from the following Gutenberg PR: https://github.com/WordPress/gutenberg/pull/48893

Trac ticket: https://core.trac.wordpress.org/ticket/58575

It basically adds style generation for link hover color to the elements block support and adds a few default values to the core theme.json files.

### CODE REVIEW

File: src/wp-includes/block-supports/elements.php [hunk 1]
```diff
@@ -104,6 +109,16 @@ function wp_render_elements_support_styles( $pre_render, $block ) {
 		)
 	);
 
+	if ( isset( $link_block_styles[':hover'] ) ) {
+		gutenberg_style_engine_get_styles(
```
Comment (peterwilsoncc) [inline src/wp-includes/block-supports/elements.php]:
```suggestion
		wp_style_engine_get_styles(
```

**Review by peterwilsoncc — COMMENTED** [review]
Does this need to support the other link pseudo classes?

https://github.com/WordPress/wordpress-develop/blob/da008c74a7117ea0ed46931b9cc8b632a09026cf/src/wp-includes/class-wp-theme-json.php#L478-L481

**Comment by youknowriad** [comment]
> Does this need to support the other link pseudo classes?

Potentially later, but for now, this just brings parity with what we offer in the Global Styles UI to the block UI.

**Comment by peterwilsoncc** [comment]
> this just brings parity with what we offer in the Global Styles UI to the block UI.

Thanks, I figured as much but I'd sure feel silly if I didn't ask ;) 

As an accessibility thing, it might be worth aliasing `:hover` to `:focus` for keyboard and assistive technology users to ensure they get visual feedback too.

If `:focus` styles are covered in another way then you can consider the PR approved (cc @tellthemachines for timezone reasons).

**Comment by youknowriad** [comment]
> As an accessibility thing, it might be worth aliasing :hover to :focus for keyboard and assistive technology users to ensure they get visual feedback too.
> If :focus styles are covered in another way then you can consider the PR approved (cc @tellthemachines for timezone reasons).

It's a great idea to nudge people to have the right accessibility guidelines applies to their themes and content but I don't think that WordPress should be making this decision for theme authors or users just like we don't do it for font sizes or things like that. We should try to show hints like we do for color contrast and headings though, that would be a great addition.

That said, this is out of scope of this PR and I'd suggest that you open an issue for it. This is merely a back port of a small change in Gutenberg to fix an inconsistency we had between the UI in global styles and the UI in the blocks. That capability to style hover colors for buttons has been on Core for some releases now.

**Comment by tellthemachines** [comment]
> If :focus styles are covered in another way 

I would consider `:focus` covered by default browser styles unless the theme is actively unsetting them (which we can't really avoid because themes can add a stylesheet with whatever styles they like in it). We don't have any way of unsetting focus styles in the UI, so users can't accidentally make their site keyboard inaccessible.

**Review by peterwilsoncc — APPROVED** [review]
Thanks both, LGTM.

**Comment by tellthemachines** [comment]
Committed in [r56028](https://core.trac.wordpress.org/changeset/56028) / [f571f45](https://github.com/WordPress/wordpress-develop/commit/f571f4519b116c4e4dcccf4d4c461b010c6a29f0).

**Pull request closed without merging**